src/database/mutations/Label.ts

```diff
@@ -25,3 +26,46 @@ export async function createLabel(
     data: { name: name, createdBy: authenticatedUser.username },
   });
 }
+
+/**
+ * @param db
+ * @param data
+ * @param authenticatedUser
+ */
+export async function updateLabel(
+  db: PrismaClient,
+  data: UpdateLabelInput,
+  authenticatedUser: AdminAPIUser
+): Promise<Label> {
+  // first check if label with updated name already exists
+  const labelNameCount = await db.label.count({
+    where: {
+      name: data.name,
+    },
+  });
+  if (labelNameCount > 0) {
+    throw new UserInputError(
+      `A label with the name "${data.name}" already exists`
+    );
+  }
+  // get count for collection-label association for label to update
+  const collectionLabelAssociationCount = await db.collection.count({
+    where: {
+      labels: {
+        some: { label: { externalId: data.externalId } },
+      },
+    },
+  });
+
+  // if there is at least one collection-label association, don't allow update
+  if (collectionLabelAssociationCount > 0) {
+    throw new UserInputError(
+      `Cannot update label; it is associated with at least one collection`
+    );
+  }
+
+  return db.label.update({
+    where: { externalId: data.externalId },
+    data: { name: data.name, updatedBy: authenticatedUser.username },
+  });
+}
```

Comment:
Nice! Good check for duplicates!

Reply:
Refactor idea: We could move the `collectionLabelAssociationCount` check to the top and do an early exit from there. That would save us the duplicate label name check because our primary case would be false anyway.